packages/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx

```diff
@@ -0,0 +1,683 @@
+    this._refArray.push({ index: legendTitle, refElement: element });
+  };
+
+  private _getCalloutContentForBar = (
+    point: IHorizontalBarChartWithAxisDataPoint,
+  ): { YValueHover: IYValueHover[]; hoverXValue: string | number | null } => {
+    const YValueHover: IYValueHover[] = [];
+    const { useSingleColor = false } = this.props;
+    const { data } = this.props;
+    const selectedPoint = data!.filter((yDataPoint: IHorizontalBarChartWithAxisDataPoint) => yDataPoint.y === point.y);
+    let selectedPointIndex = 0;
+    data!.forEach((yDataPoint: IHorizontalBarChartWithAxisDataPoint, index: number) => {
+      if (yDataPoint.y === point.y) {
+        selectedPointIndex = index;
+      }
+    });
+    let color: string;
+    if (useSingleColor) {
+      //if useSingle color , then check if user has given a palette or not
+      // and pick the first color from that or else from our paltette.
+      color = this.props.colors ? this._createColors()(1) : getNextColor(1, 0, this.props.theme?.isInverted);
+    } else {
+      color = selectedPoint[0].color
+        ? selectedPoint[0].color
+        : this.props.colors
+        ? this._createColors()(selectedPoint[0].x)
+        : getNextColor(selectedPointIndex, 0, this.props.theme?.isInverted);
+    }
+    // callout data for the bar
+    YValueHover.push({
+      legend: selectedPoint[0].legend,
+      // For HBCWA x and y Values are swapped
+      y: selectedPoint[0].x,
+      color,
+      data: selectedPoint[0].yAxisCalloutData,
+      yAxisCalloutData: selectedPoint[0].yAxisCalloutData,
+    });
+    return { YValueHover, hoverXValue: point.yAxisCalloutData || point.y.toString() };
+  };
+
+  private _onBarHover(
+    point: IHorizontalBarChartWithAxisDataPoint,
+    color: string,
+    mouseEvent: React.MouseEvent<SVGElement>,
+  ): void {
+    mouseEvent.persist();
+
+    const { YValueHover, hoverXValue } = this._getCalloutContentForBar(point);
+    if (
+      (this.state.isLegendSelected === false ||
+        (this.state.isLegendSelected && this.state.selectedLegendTitle === point.legend)) &&
+      this._calloutAnchorPoint !== point
+    ) {
+      this._calloutAnchorPoint = point;
+      this.setState({
+        refSelected: mouseEvent,
+        isCalloutVisible: true,
+        dataForHoverCard: point.x,
+        selectedLegendTitle: point.legend!,
+        color: this.props.useSingleColor ? color : point.color,
+        // To display callout value, if no callout value given, taking given point.x value as a string.
+        xCalloutValue: point.yAxisCalloutData! || point.y.toString(),
+        yCalloutValue: point.xAxisCalloutData || point.x.toString(),
+        dataPointCalloutProps: point,
+        activeXdataPoint: point.x,
+        YValueHover,
+        hoverXValue,
+        callOutAccessibilityData: point.callOutAccessibilityData,
+      });
+    }
+  }
+
+  private _onBarLeave = (): void => {
+    /**/
+  };
```

Comment:
There is a bug. Hover over a bar. Then go to a legend and select one of them. Keep changing the selection.
The hover card remains at the initially hovered bar. But on changing the bar legend selections, some data stays old and some data updates.
Lets track this as a separate bug and not hold this PR to fix it.